### PR TITLE
Travis/CI: Upgrade to xenial image

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Upgrade Travis CI from Ubuntu 14.04 to 16.04, recently added by Travis CI team. Added some comments to the pull request to explain the changes.
- Remove old unused setting to use Virtualized environment instead of Docker container as Travis CI migrated linux jobs to Docker anyway
- Reference blog post about xenial https://blog.travis-ci.com/2018-11-08-xenial-release
- Reference documentation about xenial https://docs.travis-ci.com/user/reference/xenial/
- Reference blog post about Docker forced migration https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #  (insert issue tracker number)
None

**Tests performed:** (Does it build, tested in-game, etc.)
It builds and it takes just as long as 14.04

image

**Known issues and TODO list:** (add/remove lines as needed)
- None

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,11 +10,7 @@ git:
 
 addons:
   apt:
-    sources:
-      - ubuntu-toolchain-r-test
-      - sourceline: 'ppa:kzemek/boost'
     packages:
-      - g++-6
```
Comment (jackpoz) [inline .travis.yml]: Travis comes with GCC 5.4.0 but we use clang anyway so no need to install a different GCC version

File: .travis.yml [hunk 2]
```diff
@@ -10,11 +10,7 @@ git:
 
 addons:
   apt:
-    sources:
-      - ubuntu-toolchain-r-test
-      - sourceline: 'ppa:kzemek/boost'
```
Comment (jackpoz) [inline .travis.yml]: Ubuntu 16.04 apt sources already have all the dependencies we need

**Comment by ccrs** [comment]
I'd take this opportunity to also increment the boost version used by both CI, remember my PR like a year ago?

File: .travis.yml (hunk 2 above)
Comment (ccrs) [reply .travis.yml]: I'd add here the update instruction

```
addons:
  apt:
    update: true
```

**Comment by jackpoz** [comment]
which boost version are you referring to and which feature does it have that current reqs don't ?

**Comment by ccrs** [comment]
well its about consistency I guess

https://trinitycore.atlassian.net/wiki/spaces/tc/pages/10977296/Windows+Requirements
> Boost 1.63 (latest stable recommended)

current config uses 1.58 if I'm not mistaken

**Comment by Aokromes** [comment]
linux can work with lower version than windows :)

**Comment by jackpoz** [comment]
anything against merging this ?

**Review by Shauren — APPROVED** [review]

**Pull request merged**